azurecompute-arm/src/test/java/org/jclouds/azurecomputearm/features/ResourceGroupApiMockTest.java

```diff
@@ -0,0 +1,154 @@
+import java.util.List;
+import org.jclouds.azurecomputearm.domain.ResourceGroup;
+import org.jclouds.azurecomputearm.internal.BaseAzureComputeApiMockTest;
+import org.testng.annotations.Test;
+
+@Test(groups = "unit", testName = "ResourceGroupApiMockTest", singleThreaded = true)
+public class ResourceGroupApiMockTest extends BaseAzureComputeApiMockTest {
+
+   final String subscriptionid = "12345678-1234-1234-1234-123456789012";
+   final String requestUrl = "/subscriptions/" + subscriptionid + "/resourcegroups";
+   final String version = "?api-version=2015-01-01";
+
+   public void testListResourceGroups() throws InterruptedException {
+      server.enqueue(jsonResponse("/resourcegroups.json"));
+
+      List<ResourceGroup> resourceGroups = api.getResourceGroupApi(subscriptionid).list();
+
+      assertEquals(size(resourceGroups), 2);
+
+      assertSent(server, "GET", requestUrl + version);
+   }
+
+   public void testListResourceGroupsReturns404() throws InterruptedException {
+      server.enqueue(response404());
+
+      List<ResourceGroup> resourceGroups = api.getResourceGroupApi(subscriptionid).list();
+
+      assertTrue(isEmpty(resourceGroups));
+
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", requestUrl + version);
+   }
+
+   public void testListResourceGroupsWithOptions() throws InterruptedException {
+      server.enqueue(jsonResponse("/resourcegroups.json"));
+
+      List<ResourceGroup> resourceGroups = api.getResourceGroupApi(subscriptionid).list();
+
+      assertEquals(size(resourceGroups), 2);
+      assertEquals(server.getRequestCount(), 1);
+
+      assertSent(server, "GET", requestUrl + version);
+   }
+
+   public void testListResourceGroupsWithOptionsReturns404() throws InterruptedException {
+      server.enqueue(response404());
+
+      List<ResourceGroup> resourceGroups = api.getResourceGroupApi(subscriptionid).list();
+
+      assertTrue(isEmpty(resourceGroups));
+
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "GET", requestUrl + version);
+   }
+
+   public void testCreateResourceGroup() throws InterruptedException {
+      server.enqueue(jsonResponse("/resourcegroup.json").setStatus("HTTP/1.1 201 Created"));
+
+      HashMap<String, String> tags = new HashMap<String, String>();
+      tags.put("tagname1", "tagvalue1");
+
+      ResourceGroup resourceGroup = api.getResourceGroupApi(subscriptionid).create("jcloudstest", "West US", tags);
+
+      assertEquals(resourceGroup.name(), "jcloudstest");
+      assertEquals(resourceGroup.location(), "westus");
+      assertEquals(resourceGroup.tags().size(), 1);
+      assertTrue(resourceGroup.id().contains("jcloudstest"));
+
+      assertEquals(server.getRequestCount(), 1);
+      assertSent(server, "PUT", requestUrl + "/jcloudstest" + version, String.format("{\"location\":\"%s\", \"tags\":{\"tagname1\":\"tagvalue1\"}}", "West US"));
+   }
+
+   public void testGetResourceGroup() throws InterruptedException {
```

Comment:
@andreaturli The link provided above is similar to `AzureComputeErrorHandler`. Do you have an example for `ResourceGroupParseTest`?